state_synchronizer/src/peer_manager.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::PeerId;
+use rand::{thread_rng, Rng};
+use std::collections::HashMap;
+use types::account_address::AccountAddress;
+
+#[derive(Default)]
+#[allow(dead_code)]
+pub struct PeerInfo {
+    is_alive: bool,
+    is_upstream: bool,
+}
+
+pub struct PeerManager {
+    peers: HashMap<PeerId, PeerInfo>,
+}
+
+impl PeerManager {
+    pub fn new(peer_ids: Vec<PeerId>) -> Self {
+        let peers = peer_ids
+            .into_iter()
+            .map(|peer_id| (peer_id, PeerInfo::default()))
+            .collect();
+        Self { peers }
+    }
+
+    pub fn update_peers(&mut self, peer_ids: Vec<PeerId>) {
+        let new_peers = peer_ids
+            .into_iter()
+            .map(|peer_id| (peer_id, PeerInfo::default()))
+            .collect();
+        self.peers = new_peers;
+    }
+
+    pub fn enable_peer(&mut self, peer_id: PeerId, peer_info: PeerInfo) {
+        self.peers.insert(peer_id, peer_info);
```

Comment:
enable peer shouldn't insert new entry to map
instead it should find existing entry and set is_active to true

Reply:
What if the peer wasn't in the hashmap? Should it insert + set to true if it's a new peer and set to true if existing peer that was off?

Reply:
with current approach, list of upstream peers will be known at startup time. So NewPeer event just indicates that connection is live

Reply:
I see, but self.peers.insert(peer_id, ()) is what I replaced with "enable_peer", so I'm assuming some kind of insertion is necessary still? So instead of just trying to replace a value, should I insert the peer_id with updated peer_info (with is_alive set to true)?